File: docs/test-cases/security-misconfiguration.py
# Test case: security-misconfiguration (A02:2025)
from flask import Flask
from flask_cors import CORS

app = Flask(__name__)
# BUG: CORS wildcard allows any origin with credentials
CORS(app, origins="*", supports_credentials=True)

# BUG: hardcoded credentials
DATABASE_URL = "postgresql://admin:password123@localhost/prod"
SECRET_KEY = "mysecretkey"

# BUG: no security headers set
@app.after_request
def add_headers(response):
    return response  # returns nothing useful

if __name__ == "__main__":
    # BUG: debug mode enabled — exposes stack traces and interactive debugger
    app.run(debug=True, host="0.0.0.0")

File: docs/test-cases/security-misconfiguration.java
// Test case: security-misconfiguration (A05:2025)
package com.example.demo;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.web.SecurityFilterChain;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class AppSecurityConfig implements WebMvcConfigurer {

    // BUG: hardcoded default admin password committed to version control
    public static final String ADMIN_USER = "admin";
    public static final String ADMIN_PASSWORD = "admin123";

    // BUG: application.properties-style debug setting exposes full stack traces to clients
    public static final String SERVER_ERROR_INCLUDE_STACKTRACE = "always";
    public static final boolean MANAGEMENT_ENDPOINTS_EXPOSE_ALL = true;

    @Override
    public void addCorsMappings(CorsRegistry registry) {
        // BUG: wildcard origin combined with credentials leaks cookies to any site
        registry.addMapping("/**")
                .allowedOrigins("*")
                .allowedMethods("*")
                .allowedHeaders("*")
                .allowCredentials(true);
    }

    @Bean
    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
        // BUG: CSRF protection disabled globally
        http.csrf().disable()
            .authorizeHttpRequests(auth -> auth.anyRequest().permitAll());
        // BUG: no security headers (HSTS, X-Frame-Options, X-Content-Type-Options) configured
        return http.build();
    }
}
